include/dlaf/factorization/cholesky/mc/cholesky_L.h

```diff
@@ -12,35 +12,147 @@
+      [rc_comm](hpx::shared_future<ConstTile_t> ftile, hpx::future<PromiseComm_t> fpcomm) {
+        PromiseComm_t pcomm = fpcomm.get();
+        comm::sync::broadcast::send(pcomm.ref().subCommunicator(rc_comm), ftile.get());
+      },
+      "send_tile");
+  hpx::dataflow(ex, std::move(send_bcast_f), tile, task_chain());
+}
+
+template <class T>
+hpx::future<matrix::Tile<const T, Device::CPU>> recv_tile(
+    hpx::threads::executors::pool_executor ex, common::Pipeline<comm::CommunicatorGrid>& mpi_task_chain,
+    Coord rc_comm, TileElementSize tile_size, int rank) {
+  using ConstTile_t = matrix::Tile<const T, Device::CPU>;
+  using PromiseComm_t = common::PromiseGuard<comm::CommunicatorGrid>;
+  using MemView_t = memory::MemoryView<T, Device::CPU>;
+  using Tile_t = matrix::Tile<T, Device::CPU>;
+
+  auto recv_bcast_f = hpx::util::annotated_function(
+      [rank, tile_size, rc_comm](hpx::future<PromiseComm_t> fpcomm) -> ConstTile_t {
+        PromiseComm_t pcomm = fpcomm.get();
+        MemView_t mem_view(tile_size.linear_size());
+        Tile_t tile(tile_size, std::move(mem_view), tile_size.rows());
+        comm::sync::broadcast::receive_from(rank, pcomm.ref().subCommunicator(rc_comm), tile);
+        return std::move(tile);
+      },
+      "recv_tile");
+  return hpx::dataflow(ex, std::move(recv_bcast_f), mpi_task_chain());
+}
+
+/// Updates the trailing matrix tiles of @param `mat_a` with @param `panel` column at index @param `k`.
+template <class T>
+void herkTrailingMatrix(
+    hpx::threads::executors::pool_executor ex_mpi, common::Pipeline<comm::CommunicatorGrid>& task_chain,
+    SizeType k, Matrix<T, Device::CPU>& mat_a,
+    std::unordered_map<SizeType, hpx::shared_future<matrix::Tile<const T, Device::CPU>>>& panel) {
+  using hpx::threads::executors::pool_executor;
+  using hpx::threads::thread_priority_high;
+  using hpx::threads::thread_priority_default;
+
+  pool_executor ex_hp("default", thread_priority_high);
+  pool_executor ex_normal("default", thread_priority_default);
+
+  matrix::Distribution const& distr = mat_a.distribution();
+  SizeType nrtile = mat_a.nrTiles().cols();
+  comm::Index2D this_rank = distr.rankIndex();
+
+  for (SizeType j = k + 1; j < nrtile; ++j) {
+    GlobalTileIndex jj_idx(j, j);
+    comm::Index2D jj_rank = mat_a.rankGlobalTile(jj_idx);
+
+    // Broadcast the jk-tile along the j-th column and update the jj-tile
+    if (this_rank == jj_rank) {
+      auto trailing_matrix_executor = (j == k + 1) ? ex_hp : ex_normal;
+      herk_trailing_diag_tile(trailing_matrix_executor, panel[j], mat_a(jj_idx));
+      if (j != nrtile - 1)
+        send_tile(ex_mpi, task_chain, Coord::Col, panel[j]);
+    }
+    else if (this_rank.col() == jj_rank.col()) {
+      GlobalTileIndex jk_idx(j, k);
+      if (j != nrtile - 1)
+        panel[j] = recv_tile<T>(ex_mpi, task_chain, Coord::Col, mat_a.tileSize(jk_idx), jj_rank.row());
+    }
+
+    for (SizeType i = j + 1; i < nrtile; ++i) {
```

Comment:
what about executing this loop only when I own the column?
```
if (this_rank.col() == jj_rank.col()) {
  for (....) {
     ...
  }
}
```

Reply:
@rasolca Yes, I will do something to the equivalent effect, which will avoid the nesting of the for loop:

```
if (this_rank.col() != jj_rank.col()) 
      continue;
``` 

However, I would like to know whether to drop the last commit (the refactoring of `herkTrailingMatrix()`) before that?

Reply:
Is that OK? : 1ce31d4

Reply:
LGTM.